xxhr/impl/session-curl.hpp

```diff
@@ -643,7 +643,6 @@ void Session::Impl::prepareCommon() {
 
     // Enable so we are able to retrive certificate information:
     curl_easy_setopt(curl_->handle, CURLOPT_CERTINFO, 1L);
-    curl_easy_setopt(curl_->handle, CURLOPT_NOSIGNAL, 0L);
```

Comment:
Why was it set to 0 actually ? We should avoid SIGPIPEs and disable this.

Reply:
Ok I think we should have it on, but it seems it doesn't help with SIGPIPE.

https://curl.se/libcurl/c/CURLOPT_NOSIGNAL.html

> Setting [CURLOPT_NOSIGNAL](https://curl.se/libcurl/c/CURLOPT_NOSIGNAL.html) to 1 makes libcurl NOT ask the system to ignore SIGPIPE signals, which otherwise are sent by the system when trying to send data to a socket which is closed in the other end. libcurl makes an effort to never cause such SIGPIPE signals to trigger, but some operating systems have no way to avoid them and even on those that have there are some corner cases when they may still happen, contrary to our desire.

Reply:
I think we should add a program wide asio::signal_set that ignores SIGPIPE in our app then as well.

Reply:
I can add it with 0L but in need first test with tipi because at the begin I add it with 1L and that's an error I think

Reply:
I actually think it doesn't help for SIGPIPE which we want to eliminate, so for the moment it's fine if we don't have it. 

I think global signal handling is something library code should rely on, because no one know what over lib will have for signal handling and it might conflict so generally I would disable it and have it to 1L. It seems though to make name resolvers not timeout, but our own timeout around it would manage.

Reply:
edited **NOT**

Reply:
I put it in 1L at the beginning and I think that's what's giving me the problem on linux at the moment. 
I think you've already encountered this problem in the past because you commented on it on this commit: 
https://github.com/nxxm/xxhr/commit/4a5b8671392403f6f9f9120352742c12a9aedb2d

Reply:
I was wrong in the commit you are mentioning, the doc says that it doesn't help.

Reply:
Removing this option will solve my problem with tipi.